rs/embedders/fuzz/src/imports.rs

```diff
@@ -1,94 +1,124 @@
-)
-"#;
+use std::rc::Rc;
+
+use crate::wasm_executor::{
+    get_execution_parameters, get_system_state, MAX_SUBNET_AVAILABLE_MEMORY,
+};
+use ic_config::embedders::Config as EmbeddersConfig;
+use ic_config::flag_status::FlagStatus;
+use ic_embedders::wasm_utils::instrumentation::WasmMemoryType;
+use ic_embedders::wasmtime_embedder::{system_api, StoreData};
+use ic_embedders::WasmtimeEmbedder;
+use ic_logger::replica_logger::no_op_logger;
+use ic_replicated_state::Memory;
+use ic_replicated_state::NumWasmPages;
+use ic_system_api::{ApiType, DefaultOutOfInstructionsHandler, SystemApiImpl};
+use ic_test_utilities_types::ids::user_test_id;
+use ic_types::{time::UNIX_EPOCH, NumBytes};
+use std::collections::HashMap;
+use wasm_encoder::{
+    EntityType, FuncType, ImportSection, Module, TypeSection, ValType as EncodedValType,
+};
+use wasmtime::{Engine, Extern, Store, StoreLimits, ValType};
+
+pub(crate) fn system_api_imports() -> Vec<u8> {
+    let system_api_type = WasmMemoryType::Wasm64;
+    let mut config = EmbeddersConfig::default();
+    config.feature_flags.write_barrier = FlagStatus::Enabled;
+    config.feature_flags.wasm64 = FlagStatus::Enabled;
+
+    let engine = Engine::new(&WasmtimeEmbedder::wasmtime_execution_config(&config))
+        .expect("Failed to initialize Wasmtime engine");
+    let api_type = ApiType::init(UNIX_EPOCH, vec![], user_test_id(24).get());
+
+    let canister_current_memory_usage = NumBytes::from(0);
+    let canister_current_message_memory_usage = NumBytes::from(0);
+    let system_api = SystemApiImpl::new(
+        api_type.clone(),
+        get_system_state(api_type),
+        canister_current_memory_usage,
+        canister_current_message_memory_usage,
+        get_execution_parameters(),
+        *MAX_SUBNET_AVAILABLE_MEMORY,
+        config.feature_flags.wasm_native_stable_memory,
+        config.feature_flags.canister_backtrace,
+        config.max_sum_exported_function_name_lengths,
+        Memory::new_for_testing(),
+        NumWasmPages::from(0),
+        Rc::new(DefaultOutOfInstructionsHandler::default()),
+        no_op_logger(),
+    );
+
+    let mut store = Store::new(
+        &engine,
+        StoreData {
+            system_api: Some(system_api),
+            num_instructions_global: None,
+            log: no_op_logger(),
+            num_stable_dirty_pages_from_non_native_writes: ic_types::NumOsPages::from(0),
+            limits: StoreLimits::default(),
+            canister_backtrace: config.feature_flags.canister_backtrace,
+        },
+    );
+    let mut linker: wasmtime::Linker<StoreData> = wasmtime::Linker::new(&engine);
+    system_api::syscalls::<u32>(
+        &mut linker,
+        config.feature_flags,
+        config.stable_memory_dirty_page_limit,
+        config.stable_memory_accessed_page_limit,
+        system_api_type,
+    );
+
+    // to avoid store move
+    let mut system_api_imports: Vec<(&str, &str, wasmtime::Func)> = vec![];
```

Comment:
It's a bit annoying to have the `syscalls` function public, because it's so hard to use anyway. I'd put all the code above here somewhere internal in the embedders crate (maybe it's own `mod`/file) and have it expose a public function that returns this `Vec` of available imports.

Reply:
How about I gate it with something like this. I doubt there will be many consumers of this API anyways outside fuzzing. 

```
cfg_if::cfg_if! {
    if #[cfg(feature = "fuzzing_code")] {
        pub mod system_api;
    } else {
        mod system_api;
    }
}
```

Reply:
I'm not totally convinced - won't this require recompiling the crate for fuzzing? We can talk about it today.

Reply:
Discussed offline